learn-ocaml.el

```diff
@@ -43,16 +46,21 @@
 
 (defun learn-ocaml-error-handler (buffer callback proc string)
   (if (not (string-equal string "finished\n"))
-      (message-box learn-ocaml-warning-message)
+      (when (not (string-equal string "run\n"))
+	(progn
+	  (when buffer (kill-buffer buffer))
```

Comment:
3 questions here:

* Are we sure "finished\n" and "run\n" will always end with a LF "\n", and not CR+LF "\r\n" for example? otherwise the test should be made more robust, maybe?
* what is the `buffer` involved here? more specifically, why is it necessary to do `kill-buffer buffer`
* I guess the `(if (not …) (when (not …)` could be simplified with a `(unless (or … …)`, or if you prefer, a `(case`

Reply:
when making a process a temporary buffer is created to receive the stdout of the process. So once the process is finished or it exits, it's necessary to kill the buffer to not accumulate buffers

Reply:
I think it's robust enough 
ftp://ftp.gwdg.de/pub/gnu/www/savannah-checkouts/gnu/emacs/elisp-manual/html_node/Sentinels.html actually the case "run" doesn't exit so i willl remove it

Reply:
OK, good